SmurfEvaluator/source/state_smurf/log_evaluator/LogsComparer.cpp

```diff
@@ -1,148 +1,153 @@
-		
-		if (!saveAggregatedPath.empty()) {
-			if (!saveAggregatedFiles(etalonLogs, comparedLogs, saveAggregatedPath)) {
+
+bool LogsComparer::compareFiles(std::ifstream &etalonFile, std::ifstream &comparedFile,
+								const std::string& saveAggregatedPath) {
+	CircuitAggregator circuitAggregator;
+	std::vector<std::string> etalonLogs = loadLogs(etalonFile, circuitAggregator);
+	std::vector<std::string> comparedLogs;
+	if (comparedFile.is_open()) {
+		comparedLogs = loadLogs(comparedFile, circuitAggregator);
+	} else if (saveAggregatedPath.empty()) {
+		std::cerr << "ERROR: No compare file was provided. Enter compared file or use option --save-aggregated to save aggregated etalon" << std::endl;
+		return false;
+	}
+
+	if (!saveAggregatedPath.empty()) {
+		if (comparedLogs.empty()) { 	// Creating aggregated etalon only
+			if (!helpers::FileHelper::createAggregatedFiles(etalonLogs, "etalon", saveAggregatedPath)) {
 				return false;
+			} else {
+				return true;
 			}
-		}
-		
-		if (!validateEtalon(etalonLogs)) {
-			return false;
-		}
-		if (comparedLogs.empty()) {
-			std::cerr << "ERROR: Empty Compare file" << std::endl;
+		} else if (!helpers::FileHelper::createAggregatedFiles(etalonLogs, "etalon", saveAggregatedPath) ||
+				   !helpers::FileHelper::createAggregatedFiles(comparedLogs, "compared", saveAggregatedPath)) {
 			return false;
 		}
+	}
 
-		bool filesAreSame = true;
-		bool runsAreSame = true;
-		
-		int i = 0;
-		if (!isStartOfRunLog(comparedLogs[i])) {
-			std::cerr << "ERROR: compared Log file isn't starting with Start of Run log" << std::endl;
-			return false;
-		}
-		for (int runCount = 1; i < comparedLogs.size() && isStartOfRunLog(comparedLogs[i]); ++runCount) {
-			std::cout << "Run number " << runCount << ":" << std::endl;
-			
-			for (int j = 0; j < etalonLogs.size() && i < comparedLogs.size(); ++j) {
-				if (!LineParser::compareLines(etalonLogs[j], comparedLogs[i])) {
-					runsAreSame = false;
-					
-					// Compared file is shorter than Etalon, we need to finish comparing Etalon file
-					if (isStartOfRunLog(comparedLogs[i])) {
-						if (isStartOfRunLog(etalonLogs[j])) {
-							std::cerr << "ERROR: Start of Run is different, are you comparing same application?"
-							          << std::endl;
-							return false;
-						}
-						for (j += 1; j < etalonLogs.size(); ++j) {
-							LineParser::compareLines(etalonLogs[j], "");
-						}
-						break;
+	if (!validateEtalon(etalonLogs)) {
+		return false;
+	}
+	if (comparedLogs.empty()) {
+		std::cerr << "ERROR: Empty Compare file" << std::endl;
+		return false;
+	}
+
+	bool filesAreSame = true;
+	bool runsAreSame = true;
+
+	int i = 0;
+	if (!isStartOfRunLog(comparedLogs[i])) {
+		std::cerr << "ERROR: compared Log file isn't starting with Start of Run log" << std::endl;
+		return false;
+	}
+	for (int runCount = 1; i < comparedLogs.size() && isStartOfRunLog(comparedLogs[i]); ++runCount) {
+		std::cout << "Run number " << runCount << ":" << std::endl;
```

Comment:
Why not use a Logger?

Reply:
Everywhere where std::cout, std:err

Reply:
I don't think using Logger is neccessary in this application, because it doesn't bring any information we would need. Also it would be very confusing for reading the output

Reply:
ok.